janitor/utils.py

```diff
@@ -600,3 +601,367 @@ def _complete_groupings(df, list_of_columns):
     df = df.set_index(index_columns)
 
     return df, new_reindex_columns
+
+
+def _data_checks_pivot_longer(
+    df, index, column_names, names_sep, names_pattern, names_to, values_to
+):
+
+    """
+    This function raises errors or warnings if the arguments have the wrong
+    python type, or if an unneeded argument is provided. This function is
+    executed before proceeding to the computation phase.
+    """
```

Comment:
```suggestion
    """
    This function raises errors or warnings if the arguments have the wrong
    python type, or if an unneeded argument is provided. 
    This function is executed before proceeding to the computation phase.
    
    Type annotations are not provided because this function is where type checking happens.
    """
```